### PR TITLE
Advanced configuration

### PR DESCRIPTION
**Pull request by fabien0102**
# Why

Restful-react is amazing, especially with code generation from open-api specs. But by design we have one main `Provider` that define the `base` for the backend. This is nice until we need to deal with multiple backends in the same page (yes, this kind of this happens in real projects 😅 )

The solution I would like to introduce is a way to give a `restful-react.config.js` instead of flags (as an addition of the actual usage of the CLI)

This have multiple benefits:
 - I can have only one npm command for all my backends
 - I can introduce some custom imports and props in the configuration (to have a specific `base` in the generated restful-react components)

## Example:

```js
// restful-react.config.js
module.exports = {
  myFirstBackend: {
    output: "src/queries/myFirstBackend.tsx",
    file: "specs/my-first-backend.yaml",
    customProps: {
      base: `"http://my-first-backend.com"`,
    },
  },
  configurableBackend: {
    output: "src/queries/configurableBackend.tsx",
    github: "contiamo:restful-react:master:docs/swagger.json",
    customImport: `import { getConfig } from "../components/Config.tsx";`,
    customProps: {
      base: `{getConfig("backendBasePath")}`,
    },
  },
};
```

```js
// package.json
{
  "scripts": {
    "gen": "restful-react import --config restful-react.config.js",
    "gen-first": "restful-react import --config restful-react.config.js myFirstBackend"
  }
}
```

### CODE REVIEW

File: src/bin/restful-react-import.ts [hunk 1]
```diff
@@ -2,37 +2,65 @@ import chalk from "chalk";
...
+program.option("--config [value]", "override flags by a config file");
 program.parse(process.argv);
 
-(async () => {
-  const transformer = program.transformer ? require(join(process.cwd(), program.transformer)) : undefined;
+const importSpecs = async (options: AdvancedOptions) => {
+  const transformer = options.transformer ? require(join(process.cwd(), options.transformer)) : undefined;
 
-  if (!program.output) {
+  if (!options.output) {
```
Comment (stereobooster) [inline src/bin/restful-react-import.ts]:
```ts
const { output } = options;
if (!output) {
```

File: src/scripts/import-open-api.ts [hunk 2]
```diff
@@ -227,6 +228,14 @@ const importSpecs = (data: string, extension: "yaml" | "json"): Promise<OpenAPIO
   });
 };
 
+/**
+ * Take a react props value style and convert it to object style
+ *
+ * Example:
+ *  reactPropsValueToObjectValue(`{ getConfig("myVar") }`) // `getConfig("myVar")`
+ */
+export const reactPropsValueToObjectValue = (value: string) => value.replace(/^{(.*)}$/, "$1");
```
Comment (stereobooster) [inline src/scripts/import-open-api.ts]: is there a test for it?

**Comment by stereobooster** [comment]
```
yarn examples
yarn run v1.16.0
$ run-p example:*
$ node lib/bin/restful-react.js import --github OAI:OpenAPI-Specification:master:examples/v3.0/petstore.yaml --output examples/petstoreFromGithubSpec.tsx
$ node lib/bin/restful-react.js import --config examples/restful-react.config.js
$ node lib/bin/restful-react.js import --file examples/petstore.yaml --output examples/petstoreFromFileSpec.tsx
error: unknown option `--config'
```

File: src/scripts/import-open-api.ts (hunk 2 above)
Comment (fabien0102) [reply src/scripts/import-open-api.ts]: nop ^^, but it's trivial 🙄

Comment (stereobooster) [reply src/scripts/import-open-api.ts]:
```
{`/path/${id}/test`}
```

?

Comment (fabien0102) [reply src/scripts/import-open-api.ts]: Tests added ;)

**Pull request merged**